hollow/src/main/java/com/netflix/hollow/core/read/OptionalBlobPartInput.java

```diff
@@ -0,0 +1,44 @@
+/*
+ *  Copyright 2021 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.hollow.core.read;
+
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OptionalBlobPartInput {
+
+    private final Map<String, HollowBlobInput> inputsByPartName;
+
+    public OptionalBlobPartInput() {
+        this.inputsByPartName = new HashMap<>();
+    }
+
+    public void addInput(String partName, InputStream in) {
```

Comment:
minor: I think you already considered it, but we could potentially drop this one given its in the new code path.

Reply:
I didn't add it originally, but I found myself wrapping the `InputStream` in multiple places so putting the convenience method here saved time.